interactions/context.py

```diff
@@ -574,80 +452,211 @@ async def edit(
+                    int(self.channel_id), int(self.message.id), payload=payload._json
                 )
+                self.message = Message(**res, _client=self.client)
+            else:
                 res = await self.client.edit_interaction_response(
                     token=self.token,
-                    application_id=str(self.application_id),
+                    application_id=str(self.id),
                     data={"type": self.callback.value, "data": payload._json},
+                    message_id=self.message.id if self.message else "@original",
                 )
                 if res["flags"] == 64:
                     log.warning("You can't edit hidden messages.")
+                    self.message = payload
+                    self.message._client = self.client
                 else:
                     await self.client.edit_message(
                         int(self.channel_id), res["id"], payload=payload._json
                     )
                     self.message = Message(**res, _client=self.client)
+        else:
+            res = await self.client.edit_interaction_response(
+                token=self.token,
+                application_id=str(self.application_id),
+                data={"type": self.callback.value, "data": payload._json},
+            )
+            if res["flags"] == 64:
+                log.warning("You can't edit hidden messages.")
+            else:
+                await self.client.edit_message(
+                    int(self.channel_id), res["id"], payload=payload._json
+                )
+                self.message = Message(**res, _client=self.client)
 
-        await func()
         return payload
 
+    async def defer(self, ephemeral: Optional[bool] = False) -> None:
+        """
+        This "defers" an interaction response, allowing up
+        to a 15-minute delay between invocation and responding.
+
+        :param ephemeral?: Whether the deferred state is hidden or not.
+        :type ephemeral: Optional[bool]
+        """
+        self.deferred = True
+        _ephemeral: int = (1 << 6) if ephemeral else 0
+        self.callback = InteractionCallbackType.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE
+
+        await self.client.create_interaction_response(
+            token=self.token,
+            application_id=int(self.id),
+            data={"type": self.callback.value, "data": {"flags": _ephemeral}},
+        )
+
+    async def send(self, content: str = MISSING, **kwargs):
+        payload = await super().send(content, **kwargs)
+
+        if not self.deferred:
+            self.callback = InteractionCallbackType.CHANNEL_MESSAGE_WITH_SOURCE
+
+        _payload: dict = {"type": self.callback.value, "data": payload._json}
+
+        msg = None
+        if self.responded or self.deferred:
+            if self.deferred:
+                res = await self.client.edit_interaction_response(
+                    data=payload._json,
```

Comment:
When you create the `_payload` why don't you use it here? (applies to every occurance)

Reply:
It is used in both send function, in the same way that current unstable does